src/lib/whenIdle.ts

```diff
@@ -32,7 +32,7 @@ export const whenIdle = (cb: () => void): number => {
     cb();
   } else {
     handle = rIC(cb);
-    onHidden(cb);
+    onHidden(cb, { once: true });
```

Comment:
Issue: This may not work because the `once` parameter does not take into account if the visibility state was hidden, preventing the callback from being invoked. 

```
if (document.visibilityState === 'hidden') {
      cb();
}
```